syntax = "proto3";

package inventory;

import "google/protobuf/empty.proto";

service InventoryController {
    rpc List(InventoryListRequest) returns (stream Inventory) {}
    rpc Create(Inventory) returns (Inventory) {}
    rpc Retrieve(InventoryRetrieveRequest) returns (Inventory) {}
    rpc Update(Inventory) returns (Inventory) {}
    rpc Destroy(Inventory) returns (google.protobuf.Empty) {}
}


message Inventory {
    int64 id = 1;
    string code = 2;
    enum InventoryStatus {
        CREATED = 0;
        IN_STOCK = 1;
        PROCESSING_NEEDED = 2;
        SHIPPED = 3;
        SHIPPING_PENDING = 4;
    }
    InventoryStatus status = 3;
    string product_id = 4;
    string product_name = 5;
    string product_brand_id = 6;
    string product_brand_name = 7;
    string product_size = 8;
    string product_color = 9;
    string location = 10;
    string memo = 11;
    string created_at = 12;
    string updated_at = 13;
    optional string deleted_at = 14;
}

message InventoryListRequest {
    optional string code = 1;
    optional string product_name = 2;
    optional string product_brand_name = 3;
}

message InventoryRetrieveRequest {
    int64 id = 1;
}
